### ISSUE TITLE
e_cabf_ecc_allowed_key_usages: false positive on CA certificates

### ISSUE THREAD

**Issue by tsaarni** [zmap/zlint#1050]
The `e_cabf_ecc_allowed_key_usages` lint (introduced in [v3.7.1](https://github.com/zmap/zlint/releases/tag/v3.7.1), #1041) fires on CA certificates, but the requirement seems to apply only to Subscriber certificates.

Example: an ECC root CA with `keyCertSign` | `cRLSign`

```json
{
  "e_cabf_ecc_allowed_key_usages": {
    "result": "error",
    "details": "DigitalSignature key usage is required for certificates with ECC public keys"
  }
}
```
The CA/B Forum BRs has separate key usage requirements for subscribers vs CAs:

**7.1.2.7.11 Subscriber Certificate Key Usage ([link](https://github.com/cabforum/servercert/blob/main/docs/BR.md#712711-subscriber-certificate-key-usage))**

Table: Key Usage for ECC Public Keys

| Key Usage | Permitted | Required | 
| - | - | - | 
| digitalSignature | Y | MUST | 

**7.1.2.10.7 CA Certificate Key Usage ([link](https://github.com/cabforum/servercert/blob/main/docs/BR.md#712107-ca-certificate-key-usage))**

| Key Usage | Permitted | Required |
| - | - | - |
| digitalSignature | Y | N <sup>footnote</sup> |

where reference footnote is:
> If a CA Certificate does not assert the digitalSignature bit, the CA Private Key MUST NOT be used to sign an OCSP Response. See [Section 7.3](https://github.com/cabforum/servercert/blob/main/docs/BR.md#73-ocsp-profile) for more information.

The lint's `CheckApplies` does not exclude CA certificates ([link](https://github.com/zmap/zlint/blob/ec384e20d187feb9b2411ced4a1ba7315bb0555e/v3/lints/cabf_br/lint_ecc_allowed_key_usages.go#L65-L67)).

**Comment by tsaarni** [comment]
hi @digirenpeter, could you please check this issue when you have time?

**Comment by digirenpeter** [comment]
Hey @tsaarni, that's an oversight on my part. I've opened [this PR](https://github.com/zmap/zlint/pull/1051) to fix the lint so it only triggers on subscriber certificates. Thanks for pointing it out!

**Comment by tsaarni** [comment]
Thank you @digirenpeter!
